.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -119,6 +119,9 @@ variables:
   NUGET_ENABLE_EXPERIMENTAL_HTTP_RETRY: true
   DefaultTimeout: 60
   DD_INSTRUMENTATION_TELEMETRY_ENABLED: 0
+  DD_LOGGER_SERVICE: dd-trace-dotnet
+  DD_LOGGER_ENV: CI
+  DD_LOGGER_API_KEY: $(DD_API_KEY)
```

Comment:
Setting these here means they're defined for _all_ the jobs, and are passed in even to the build steps etc. I don't know if you want to do that? 🤔

Reply:
No, `DD_API_KEY` is a secret, and secrets are not mapped to env-vars automatically, we have to specify an `env` in every step that uses it.